src/grid/index.tsx

```diff
@@ -164,7 +165,10 @@ export default class Grid<S> extends I18nMixin(ThemedMixin(WidgetBase))<GridProp
 	}
 
 	private _pageChange(page: number) {
-		const { storeId } = this._getProperties();
+		const { pagination, storeId } = this._getProperties();
+		if (pagination) {
+			clearSelectionProcess(this._store)({ id: storeId });
+		}
```

Comment:
This removes selection when changing pages on paginated grids. This was an issue also reported on #1436.